public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -59,8 +65,12 @@ export class AppChromeService {
 
   public update(update: Partial<AppChromeState>) {
     const current = this.state.getValue();
+    const sessionStorageReturnToPrevious = window.sessionStorage.getItem('returnToPrevious');
```

Comment:
do you need this anymore now that update is always called with the new `returnToPrevious` value?

Reply:
I added this in order to get the data from `sessionStorage` when the user refreshes the page. Is this not right? 🤔